tchannel-core/src/main/java/com/uber/tchannel/api/handlers/JSONRequestHandler.java

```diff
@@ -20,22 +20,51 @@
  * THE SOFTWARE.
  */
 
-package com.uber.tchannel.api;
+package com.uber.tchannel.api.handlers;
 
 import com.google.common.reflect.TypeToken;
+import com.uber.tchannel.api.ResponseCode;
+import com.uber.tchannel.schemes.JSONSerializer;
+import com.uber.tchannel.schemes.RawRequest;
+import com.uber.tchannel.schemes.RawResponse;
 
-public abstract class DefaultRequestHandler<T, U> implements RequestHandler<T, U> {
+import java.util.Map;
+
+public abstract class JSONRequestHandler<T, U> implements RequestHandler {
 
     private TypeToken<T> requestType = new TypeToken<T>(getClass()) { };
     private TypeToken<U> responseType = new TypeToken<U>(getClass()) { };
 
+    private JSONSerializer serializer = new JSONSerializer();
+
     @Override
+    public RawResponse handle(RawRequest request) {
+
+        String endpoint = serializer.decodeEndpoint(request.getArg1());
+        Map<String, String> headers = serializer.decodeHeaders(request.getArg2());
+        T param = serializer.decodeBody(request.getArg3(), this.getRequestType());
+
+        U response = handleImpl(param);
+
+        RawResponse rawResponse = new RawResponse(
+                request.getId(),
+                ResponseCode.OK,
+                request.getTransportHeaders(),
+                serializer.encodeEndpoint(endpoint),
+                serializer.encodeHeaders(headers),
+                serializer.encodeBody(response)
+        );
+
+        return rawResponse;
+    }
+
     public Class<T> getRequestType() {
         return (Class<T>) requestType.getRawType();
     }
 
-    @Override
     public Class<U> getResponseType() {
         return (Class<U>) responseType.getRawType();
     }
+
+    public abstract U handleImpl(T params);
```

Comment:
want to switch the naming here with `handle` I feel like it's cleaner for clients to implement `handle`. Also, can we make this a required method for clients?